dts/bindings/pinctrl/nxp,mcux-rt-pinctrl.yaml

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2021 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  The node has the 'pinctrl' node label set in MCUX RT SoC's devicetree. These
+  node can be autogenerated using the MCUXpresso config tools combined with
+  the parse-rt-mex.py script in the NXP HAL.
+
+
+compatible: "nxp,mcux-rt-pinctrl"
+
+include:
+    - name: base.yaml
+    - name: pincfg-node-group.yaml
+      child-binding:
+        property-allowlist:
+          - drive-open-drain
+
+properties:
+    reg:
+      required: true
+
+child-binding:
+  description: |
+    MCUX RT pin controller pin configuration group.
+  child-binding:
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of values defining the configuration register for a given
+          pinmux mode. See fsl_iomuxc.h in the NXP HAL for a list.
+      nxp,software_input_on:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          00 Pin input follows peripheral
+          01 Pin is forced to input
+      nxp,hysteresis_enable:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          00 Hyst. Disable Field
+          01 Hyst. Enable Field
+      nxp,pull_up_down_config:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+          - 2
+          - 3
+        description: |
+          00 PUS_0_100K_Ohm_Pull_Down — 100K Ohm Pull Down
+          01 PUS_1_47K_Ohm_Pull_Up — 47K Ohm Pull Up
+          10 PUS_2_100K_Ohm_Pull_Up — 100K Ohm Pull Up
+          11 PUS_3_22K_Ohm_Pull_Up — 22K Ohm Pull Up
+      nxp,pull_keeper_enable:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          0 PKE_0_Pull_Keeper_Disabled — Pull/Keeper Disabled
+          1 PKE_1_Pull_Keeper_Enabled — Pull/Keeper Enabled
+      nxp,pull_keeper_select:
+        required: false
+        type: int
+        default: 0
+        enum:
+          - 0
+          - 1
+        description: |
+          0 PUE_0_Keeper — Keeper
+          1 PUE_1_Pull — Pull
+      nxp,drive_strength:
```

Comment:
standard `drive-strength` can be used (you can customize it as ST/GD does for `slew-rate`)

Reply:
it is already customried, and no other platfrom can reuse this rule. I would rather to have NXP dedicated naming, if it is only apply to NXP platform. but leave @danieldegrasse to decide.